src/Adapter/Order/Refund/OrderRefundCalculator.php

```diff
@@ -81,49 +81,58 @@ public function computeOrderRefund(
-            $refundedAmount -= $voucherAmount = (float) $order->total_discounts;
+            $voucherAmount = new Number((string) $order->total_discounts);
+            $refundedAmount = $refundedAmount->minus($voucherAmount);
         } elseif ($voucherRefundType === VoucherRefundType::SPECIFIC_AMOUNT_REFUND) {
             $voucherChosen = true;
             $refundedAmount = $voucherAmount = $chosenVoucherAmount;
         }
 
-        $shippingCostAmount = (float) (string) $shippingRefund;
-        if ($shippingCostAmount > 0) {
-            $shippingMaxRefund = $isTaxIncluded ? $order->total_shipping_tax_incl : $order->total_shipping_tax_excl;
+        $shippingCostAmount = new Number((string) ($shippingRefund ?? 0));
+        if ($shippingCostAmount->isPositive()) {
+            $shippingMaxRefund = new Number(
+                $isTaxIncluded ?
+                    (string) $order->total_shipping_tax_incl :
+                    (string) $order->total_shipping_tax_excl
+            );
+
             $shippingSlipResume = OrderSlip::getShippingSlipResume($order->id);
-            $shippingMaxRefund -= $shippingSlipResume['total_shipping_tax_incl'] ?? 0;
-            if ($shippingCostAmount > $shippingMaxRefund) {
+            $shippingSlipTotalTaxIncl = new Number((string) ($shippingSlipResume['total_shipping_tax_incl'] ?? 0));
+            $shippingMaxRefund = $shippingMaxRefund->minus($shippingSlipTotalTaxIncl);
+
+            if ($shippingCostAmount->isGreaterThan($shippingMaxRefund)) {
                 $shippingCostAmount = $shippingMaxRefund;
             }
             if (!$isTaxIncluded) {
-                // @todo: use https://github.com/PrestaShop/decimal for price computations
                 $taxCalculator = $this->getCarrierTaxCalculatorFromOrder($order);
-                $refundedAmount += $taxCalculator->addTaxes($shippingCostAmount);
+                $taxesAmount = $taxCalculator->addTaxes((float) (string) $shippingCostAmount);
+                $taxes = new Number((string) $taxesAmount);
+                $refundedAmount = $refundedAmount->plus($taxes);
             } else {
-                $refundedAmount += $shippingCostAmount;
+                $refundedAmount = $refundedAmount->plus($shippingCostAmount);
             }
         }
 
         // Something has to be refunded (check refunds count instead of the sum in case a voucher is implied)
-        if (count($productRefunds) <= 0 && $shippingCostAmount <= 0) {
+        if (count($productRefunds) <= 0 && $refundedAmount->isLowerOrEqualThan(new Number('0'))) {
             throw new InvalidCancelProductException(InvalidCancelProductException::NO_REFUNDS);
         }
 
         return new OrderRefundSummary(
             $orderDetailList,
             $productRefunds,
-            $refundedAmount,
-            $shippingCostAmount,
-            $voucherAmount,
+            (float) (string) $refundedAmount,
```

Comment:
Not in this PR but maybe those should be `Number` as well (in a further refacto)

Reply:
yeah I guess, this one might have bigger scope, lets leave it for another time